app/code/Magento/SalesRule/Model/ResourceModel/Report/Collection.php

```diff
@@ -41,27 +49,27 @@ class Collection extends \Magento\Sales\Model\ResourceModel\Report\Collection\Ab
     protected $_rulesIdsFilter;
 
     /**
-     * @var \Magento\SalesRule\Model\ResourceModel\Report\RuleFactory $ruleFactory
+     * @var RuleFactory $ruleFactory
      */
     protected $_ruleFactory;
 
     /**
-     * @param \Magento\Framework\Data\Collection\EntityFactory $entityFactory
-     * @param \Psr\Log\LoggerInterface $logger
-     * @param \Magento\Framework\Data\Collection\Db\FetchStrategyInterface $fetchStrategy
-     * @param \Magento\Framework\Event\ManagerInterface $eventManager
-     * @param \Magento\SalesRule\Model\ResourceModel\Report\RuleFactory $ruleFactory
-     * @param \Magento\Sales\Model\ResourceModel\Report $resource
+     * @param EntityFactory $entityFactory
+     * @param LoggerInterface $logger
+     * @param FetchStrategyInterface $fetchStrategy
+     * @param ManagerInterface $eventManager
+     * @param RuleFactory $ruleFactory
+     * @param ResourceReport $resource
      * @param mixed $connection
      */
     public function __construct(
-        \Magento\Framework\Data\Collection\EntityFactory $entityFactory,
-        \Psr\Log\LoggerInterface $logger,
-        \Magento\Framework\Data\Collection\Db\FetchStrategyInterface $fetchStrategy,
-        \Magento\Framework\Event\ManagerInterface $eventManager,
-        \Magento\Sales\Model\ResourceModel\Report $resource,
-        \Magento\SalesRule\Model\ResourceModel\Report\RuleFactory $ruleFactory,
-        \Magento\Framework\DB\Adapter\AdapterInterface $connection = null
+        EntityFactory $entityFactory,
+        LoggerInterface $logger,
+        FetchStrategyInterface $fetchStrategy,
+        ManagerInterface $eventManager,
+        ResourceReport $resource,
+        RuleFactory $ruleFactory,
```

Comment:
Why not promote this property?
```suggestion
        protected readonly RuleFactory $_ruleFactory,
```